packages/loopback/test/acceptance/context-bindings/class-level-bindings.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from 'testlab';
+import {Context, inject} from '../../../lib/context';
+
+describe.only('Context bindings - Injecting dependencies of classes', () => {
+  let ctx: Context;
+  before('given a context', createContext);
+
+  it('injects constructor args', () => {
+    ctx.bind('application.name').to('CodeHub');
+
+    class InfoController {
+      constructor(@inject('application.name') public appName: string) {
+      }
+    }
+    ctx.bindClass('controllers.info', InfoController);
```

Comment:
This doesn't really follow the convention of `bind()` + `to()`. If we really need `bindClass` I would expect it to be like this:

```ts
ctx.bind('controllers.info').toClass(InfoController);
// or
ctx.bind('controllers.info').toType(InfoController);
```